massa-execution-worker/src/tests/scenarios_mandatories.rs

```diff
@@ -1,3115 +1,2484 @@
+        &block_producer,
+        Slot::new(1, 0),
+        vec![operation],
+        vec![],
+        vec![],
+    );
+    // store the block in storage
+    universe.storage.store_block(block.clone());
+
+    // set our block as a final block
+    let mut finalized_blocks: HashMap<Slot, BlockId> = Default::default();
+    finalized_blocks.insert(block.content.header.content.slot, block.id);
+    let block_store = vec![(
+        block.id,
+        ExecutionBlockMetadata {
+            storage: Some(universe.storage.clone()),
+            same_thread_parent_creator: Some(Address::from_public_key(&keypair.get_public_key())),
+        },
+    )]
+    .into_iter()
+    .collect();
+    universe.module_controller.update_blockclique_status(
+        finalized_blocks,
+        Default::default(),
+        block_store,
+    );
+    std::thread::sleep(
+        exec_cfg
+            .t0
+            .saturating_add(MassaTime::from_millis(50))
+            .into(),
+    );
+
+    let events = universe
+        .module_controller
+        .get_filtered_sc_output_event(EventFilter::default());
+    // match the events
+    assert_eq!(events.len(), 2);
+    assert!(
+        events[0].data.ends_with("true"),
+        "Expected 'true': {:?}",
+        events[0].data
+    );
+    assert!(
+        events[1].data.ends_with("false"),
+        "Expected 'false': {:?}",
+        events[1].data
+    );
+
+    assert_eq!(
+        universe
+            .final_state
+            .read()
+            .ledger
+            .get_balance(&Address::from_public_key(&keypair.get_public_key()))
+            .unwrap(),
+        Amount::from_str("300000")
+            .unwrap()
+            // Gas fee
+            .saturating_sub(Amount::const_init(10, 0))
+    );
+}
 
-    #[test]
-    fn test_take_batch() {
-        let final_state = get_sample_state(0).unwrap().0;
-        let active_history = Arc::new(RwLock::new(ActiveHistory::default()));
-
-        let mut speculative_pool = SpeculativeAsyncPool::new(final_state, active_history);
-
-        let address = Address::User(UserAddress::UserAddressV0(UserAddressV0(
-            Hash::compute_from(b"abc"),
-        )));
-
-        for i in 1..10 {
-            let message = AsyncMessage::new(
-                Slot::new(0, 0),
-                0,
-                address,
-                address,
-                "function".to_string(),
-                i,
-                Amount::from_str("0.1").unwrap(),
-                Amount::from_str("0.3").unwrap(),
-                Slot::new(1, 0),
-                Slot::new(3, 0),
-                Vec::new(),
-                None,
-                None,
-            );
-            speculative_pool.push_new_message(message)
-        }
-        assert_eq!(speculative_pool.get_message_infos().len(), 9);
-        speculative_pool.take_batch_to_execute(Slot::new(2, 0), 19);
-        assert_eq!(speculative_pool.get_message_infos().len(), 4);
-    }
+#[test]
+fn test_rewards() {
```

Comment:
unless I'm mistaken, you only give the rewards to one address in this test? It would be more comprehensive to also test the splits (endorsement, block creator, block creators of endorsed blocks).

Reply:
Ok will change it

Reply:
Done